### ISSUE TITLE
Changing branch name prefix

### ISSUE THREAD

**Issue by qeroll** [atlassian/github-for-jira#1886]
Hi,

I have integrated Jira with GitHub in order to create branches from Jira.
However, when I use issue_type as part of the branch name it can appear only as bug/ or task/. I would like it to be bugfix for Bugs and feature for Tasks / Story.

Is that at all possible?

**Comment by Ryan-Yuanqing-Jiang** [comment]
Hi, thanks for using the Github-for-Jira app and reaching out to us! 

Just to make sure I understand it correctly: Do you mean if it's possible to change the Jira issue's type according to the branch name prefix entered during creating branches from Jira issue? 

If so, we don't currently support this functionality and it's not on our near term roadmap yet. I will keep a note of this in our future feature-requests.

Thanks!

**Comment by qeroll** [comment]
Hi,

I think you might have understood me correctly but just to make sure you do, I'll explain in more detail:

* After adding the GitHub - Jira App I can use the "Create Branch" button directly from the Jira Issue
* When I press the button I can change the format of the branch name to: issue_type/issue_key-issue_summary
* I would like to be able to decide to which text, issue_type translates. For example: task is now "task". I want task to be "feature". Bug should be "bugfix" and not bug and so on.

Is that an option or the string has to be the default which is defined in Jira?

Thanks in advance!

**Comment by Ryan-Yuanqing-Jiang** [comment]
Thanks for clarifying it!

At the moment we don't support this functionality yet, so it'd be a manual step to change the branch name for the time being.
But I will triage this into our feature request backlog.

Cheers.

**Comment by gbsmith** [comment]
Second this!

**Comment by Skyppid** [comment]
Would be great, we use Git Workflow and need to name branches like this `feature/{story-no}/{story-title}` for stories and `hotfix/{bug-no}/{bug-title}` for bugs. Best would be `feature/{story-no}/{sub-task-no}/{task-title}` for nested tickets.

So a bit more flexibility when configuring the scheme would be nice!

**Comment by jkoelle** [comment]
Echoing what others have said. We use Git Workflow and it would be helpful for the prefix to be added to the branch name when creating from Jira so that developers don't need to do this manually each time they create a branch.

**Comment by jaredtbates** [comment]
Agreed, this is something our team is used to and it would save us time if we could automate the prefixing of branch names.

**Comment by rudacs** [comment]
I agree.
The goal would be to standardize.

Vote +1

**Comment by urshess-siga** [comment]
I agree as well. This is the way the Bitbucket integration works, btw. 
feature/{story-no}/{story-title} for stories and 
hotfix/{bug-no}/{bug-title} for bugs.
So you don't even have to "re-invent the wheel"

**Comment by yribeiro-sent** [comment]
This would be ideal for me and my team! +1 vote!

**Comment by kris-skillprofile** [comment]
We need this too!

**Comment by stan-spotts** [comment]
Any status here? Gitflow is probably the most popular workflow in Git, and it's surprising this wasn't supported right out of the box. We're using it, like many others, and it would help a lot if this is supported.

**Comment by Skyppid** [comment]
Well we have found a simple solution: Likely migrating to Github Projects instead of JIRA. It held it's title long enough but the lack of support and the ever more cracking foundation of this monolith makes the newcomers shine a lot more. So many parts of JIRA get no love anymore since Atlassian seems to focus on cash milking their other cows nowadays... big company things.

**Comment by ribase** [comment]
Is there anything going on in that direction?
